pkg/Cpanel/Security/Advisor/Assessors/Kernel.pm

```diff
@@ -29,22 +29,142 @@ package Cpanel::Security::Advisor::Assessors::Kernel;
+    if ( Cpanel::Version::compare( Cpanel::Version::getversionnumber(), '>=', $KC_CP_VERSION ) ) {
+        $self->_suggest_kernelcare;
+    }
+
     $self->_check_for_kernel_version;
 
     return 1;
 }
 
+sub _suggest_kernelcare {
+    my ($self) = @_;
+
+    my $environment  = Cpanel::OSSys::Env::get_envtype();
+    my $manage2_data = _get_manage2_kernelcare_data();
+    my $rpm          = Cpanel::RPM->new();
+
+    if (    not $rpm->has_rpm(q{kernelcare})
+        and not( $environment eq 'virtuozzo' || $environment eq 'lxc' )
+        and not $manage2_data->{'disabled'} ) {
+
+        my $contact_method = '';
+        my $target         = '_parent';
+        my $url_to_use     = $self->base_path('scripts12/purchase_kernelcare_init');
+
+        # check to see this IP has a valid license even if it is not installed
+        if ( _verify_kernelcare_license() ) {
+            my $url_alt_text = 'Click to install';
+            $url_to_use = $self->base_path('scripts12/purchase_kernelcare_completion?order_status=success');
+
+            $self->add_bad_advice(
+                'key'        => 'Kernel_kernelcare_valid_license_but_not_installed',
+                'text'       => $self->_lh->maketext('Valid KernelCare License Found, but KernelCare is Not Installed.'),
+                'suggestion' => $self->_lh->maketext( 'KernelCare provides an easy and effortless way to ensure that your operating system uses the most up-to-date kernel without the need to reboot your server. [_1] [output,url,_2,_3,_4,_5].', $contact_method, $url_to_use, $url_alt_text, 'target', $target, ),
+            );
+        }
+        else {
+            my $url_alt_text = 'Upgrade to KernelCare';
+            if ( $manage2_data->{'url'} ne '' ) {
+                $url_to_use = $manage2_data->{'url'};
+            }
+            elsif ( $manage2_data->{'email'} ne '' ) {
+                $target         = '_blank';
+                $url_to_use     = 'mailto:' . $manage2_data->{'email'};
+                $contact_method = 'For more information,';
+                $url_alt_text   = 'email your provider';
+            }
+            $self->add_warn_advice(
+                'key'        => 'Kernel_kernelcare_purchase',
+                'text'       => $self->_lh->maketext('Upgrade to KernelCare.'),
+                'suggestion' => $self->_lh->maketext( 'KernelCare provides an easy and effortless way to ensure that your operating system uses the most up-to-date kernel without the need to reboot your server. [_1] [output,url,_2,_3,_4,_5].', $contact_method, $url_to_use, $url_alt_text, 'target', $target, ),
+            );
+
+        }
+    }
+
+    return 1;
+}
+
+sub _verify_kernelcare_license {
+    my $mainserverip = Cpanel::NAT::get_public_ip( Cpanel::DIp::MainIP::getmainserverip() );
+    my $verify_url   = sprintf( "%s/ipaddrs.cgi?ip=%s", $KC_VERIFY_URL, $mainserverip );
+    my $response     = HTTP::Tiny->new->get($verify_url);
```

Comment:
You either need to use Cpanel::HTTP::Client or explicitly enable SSL verification here, since HTTP::Tiny doesn't do it by default.